Sources/OktaOAuth2/Authentication/SessionTokenFlow.swift

```diff
@@ -0,0 +1,272 @@
+//
+// Copyright (c) 2022-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+import AuthFoundation
+
+/// An authentication flow class that exchanges a Session Token for access tokens.
+public class SessionTokenFlow: AuthenticationFlow {
```

Comment:
I haven't been marking the classes as `final` since some of them required subclassing to override behaviours in tests. You bring up a good point though, since after auditing my code, it doesn't appear I need to do that anymore.